utils/frame/try-runtime/cli/src/commands/benchmark_upgrade.rs

```diff
@@ -0,0 +1,119 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::{fmt::Debug, str::FromStr, time};
+
+use parity_scale_codec::Decode;
+use sc_executor::NativeExecutionDispatch;
+use sc_service::Configuration;
+use sp_runtime::traits::{Block as BlockT, NumberFor};
+
+use crate::{
+	build_executor, ensure_matching_spec, extract_code, local_spec, state_machine_call,
+	SharedParams, State, LOG_TARGET,
+};
+
+/// Configurations of the [`Command::BenchmarkUpgradeCmd`].
+#[derive(Debug, Clone, structopt::StructOpt)]
+pub struct BenchmarkUpgradeCmd {
+	/// The state type to use.
+	#[structopt(subcommand)]
+	pub state: State,
+
+	/// Number of iterations of _`on_runtime_upgrade()`
+	#[structopt(long)]
+	steps: u64,
+}
+
+pub(crate) async fn on_runtime_upgrade_bench<Block, ExecDispatch>(
+	shared: SharedParams,
+	command: BenchmarkUpgradeCmd,
+	config: Configuration,
+) -> sc_cli::Result<()>
+where
+	Block: BlockT + serde::de::DeserializeOwned,
+	Block::Hash: FromStr,
+	<Block::Hash as FromStr>::Err: Debug,
+	NumberFor<Block>: FromStr,
+	<NumberFor<Block> as FromStr>::Err: Debug,
+	ExecDispatch: NativeExecutionDispatch + 'static,
```

Comment:
so that mean we are benchmarking the native execution ? not the wasm one ?

Reply:
No the executor is `NativeElseWasmExecutor`, giving the option for either. So in my example output used `--execution wasm`. But that is a good point I should make it so native execution is not an option